apis/r/tests/testthat/test-query-condition.R

```diff
@@ -0,0 +1,206 @@
+test_that("DataFrame Factory", {
```

Comment:
Some other cases I can think of:
- Mismatched types like `soma_joinid == "ten"`
- What is the behavior for comparing floats and strings like `soma_joinid == 10.0`?
- What is the behavior for overflow/underflow  `int8== 100000000`? 
- Comparing signed and unsigned `int8 == -1`
- Make sure parens evaluate correctly
- Incomplete query conditions: `soma_joinid ==`, `soma_joinid && int8`, `soma_joinid == 1 &&` 

Could potentially do this on a follow-up PR.